org.eclipse.buildship.core.test/build.gradle

```diff
@@ -13,3 +15,35 @@ eclipseTest {
     optionsFile rootProject.project(':org.eclipse.buildship.core').file('.options')
 }
 
+def compatibilityCheckClassName = 'org.eclipse.buildship.core.internal.workspace.ImportingProjectCrossVersionTest'
+def compatibilityCheckDescriptorPrefix = 'Can import a simple project build with Gradle '
+
+tasks.getByName("crossVersionEclipseTest") {
```

Comment:
You're right, and IMO it depends on when we will update the compatibility matrix.

The existing matrix (for current version, the JVM 8~17) should be strictly "final" and is expected not to be edited any more, since it's for published JDK and Gradle versions. They can be checked for the current CI process as you mentioned.

But for a new JDK version, e.g., we're testing the lowest Gradle supported version for JDK 18,  If we return when a test case fails, how can we find the expected version?

Reply:
A quick idea is we can update this matrix when all tests are finished, not per test case (this PR currently uses), in this way, we can choose to add a new property in the matrix or check an existing property. But this change could happen in the CI machine, how to update this matrix is still a problem.

Reply:
IIUC we need to keep the CI up-to-date with the JDK versions. I can also imagine to add an extra cross-version coverage for early access JDK releases.

Reply:
So I can imagine the workflow is:

- In this PR, we will initialize a compatibility matrix conatins JDK 7~18
- if the CI result does not match the current compatibility matrix, it will report the actual result and the expect result
- When JDK 19 is (early access) available, we will add related JDK info and cross-version coverage, at the first time, the compatibility information of JDK 19 has been generated in CI machine, we need to manually get it and update it to the master branch
- For the following CI runs, it will normally check JDK 7~19

does this make sense?

Reply:
Sorry for not replying earlier. Yes, that makes sense!